daml/Marketplace/Trading/Service.daml

```diff
@@ -1,99 +1,85 @@
-
+      unlockDeposit : ContractId AssetDeposit -> Update ()
+      unlockDeposit = \depositCid -> exercise depositCid AssetDeposit_Unlock >> pure ()
+      unlockCollateral : Order.MarketType -> Update ()
+      unlockCollateral = \case
+        Order.Collateralized depositCid _ -> unlockDeposit depositCid
+        _ -> pure ()
 
     controller customer can
-      nonconsuming RequestCreateOrder : Either (ContractId Order.T)
-                                               (ContractId Order.T, ContractId CreateOrderRequest, Order.TradeCollateral)
+      nonconsuming RequestCreateOrder : ContractId Order.T
         with
-          collateral : Order.TradeCollateral
           details : Order.Details
         do
+          createdAt <- getTime
+
           let
             status = Order.New
             providerOrderId = None
             executions = []
+            observers = Set.empty
+            collateral = None
             remainingQuantity = details.asset.quantity
-
-          let detailsWithDepositedFees : Update Order.Details
-              detailsWithDepositedFees = case details.optExchangeFee of
-                  Some feeDepositCid -> do
-                    newDepositCid <- exerciseByKey @AllocationAccountRule.T allocationAccount.id
-                                        AllocationAccountRule.Deposit with depositCid = feeDepositCid; ..
-                    return details with optExchangeFee = Some newDepositCid
-                  None -> return details
-
-          createdAt <- getTime
-
-          case collateral of
-            (Order.Collateral depositCid) -> do
-                details <- detailsWithDepositedFees
-                collateral <- Order.Collateral <$> exerciseByKey @AllocationAccountRule.T allocationAccount.id AllocationAccountRule.Deposit with ..
-                createOrderRequestCid <- create CreateOrderRequest with ..
-                orderCid <- create Order.Order with ..
-
-                return $ Right (orderCid, createOrderRequestCid, collateral)
-
-            (Order.Cleared clearinghouse) -> do
-              optClearingService <- lookupByKey @Clearing.Service (operator,clearinghouse,customer)
-              case optClearingService of
-                None -> do
-                  let reason = Error with code = "3001"; message = "Customer is not a member of clearinghouse"
-                  Left <$> create Order.Order with status = Order.Rejected{reason}; ..
-
-                (Some serviceCid) -> exercise serviceCid Clearing.ApproveTrade >>= \case
+            detailsWithDepositedFees : Update Order.Details
+            detailsWithDepositedFees = case details.optExchangeFee of
+              Some feeDepositCid -> do
+                newDepositCid <- exercise feeDepositCid AssetDeposit_Lock with newLockers = Set.fromList [ provider ]
+                return details with optExchangeFee = Some newDepositCid
```

Comment:
Do I understand this correctly here: we lock the entire fee amount that the user has provided, and then unlock and transfer parts of it with every execution?

I don't see anywhere that we check the provided fee amount is actually sufficient to execute the full order. Don't need to add this in this PR, but maybe open a ticket for it.

Reply:
This is currently checked in the Exchange Adapter / Matching Engine. The reason for that as a workaround was without `submitMulti`, the `customer` exercising this choice has no way of being able to look up the `FeeSchedule` to get the fee amount.

The check can be moved here when we can start using `submitMulti` with Public party reads. I know that @alexmatson-da has been working on supporting this in hub - is it integrated into the frontend yet?

Reply:
Yep being able to retrieve a token with `readAs: public-id` is now supported in Hub, which should allow the submitMulti. It's not yet integrated into the marketplace frontend code, though